src/APIM_ARMTemplate/apimtemplate/Extractor/EntityExtractors/TagExtractor.cs

```diff
@@ -8,27 +8,59 @@
 
 namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extract
 {
-    public class TagExtractor: EntityExtractor
+    public class TagExtractor : EntityExtractor
     {
-        public async Task<string> GetTagsAsync(string ApiManagementName, string ResourceGroupName)
+        
+        /// <summary>
+        /// Creates the Tags request URL based on the Azure subscription ID, Resource Group name, and APIM instance name.
+        /// </summary>
+        /// <param name="ApiManagementName"></param>
+        /// <param name="ResourceGroupName"></param>
+        /// <returns>A string representing a request URL</returns>
+        /// <remarks>
+        /// This was split out from the GetTagsAsync method in order to make repeated calls with "nextLink" URLs.
+        /// </remarks>
+        protected async Task<string> GetBaseTagsUrlAsync(string ApiManagementName, string ResourceGroupName)
         {
             (string azToken, string azSubId) = await auth.GetAccessToken();
 
-            string requestUrl = string.Format("{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/Tags?api-version={4}",
+            return string.Format("{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/Tags?api-version={4}",
                baseUrl, azSubId, ResourceGroupName, ApiManagementName, GlobalConstants.APIVersion);
+        }
 
+        /// <summary>
+        /// Makes an Azure API Management REST API call
+        /// </summary>
+        /// <param name="requestUrl"></param>
+        /// <returns>a JSON string with the API call results</returns>
+        /// <remarks>
+        /// Request URL creation was moved out of here in order to make repeated calls to this method with different URLs.
+        /// Changed to protected because I didn't see a reason for it to be public.
+        /// </remarks>
+        protected async Task<string> GetTagsAsync(string requestUrl)
+        {
+            (string azToken, string azSubId) = await auth.GetAccessToken();
             return await CallApiManagementAsync(azToken, requestUrl);
         }
 
-        public async Task<Template> GenerateTagsTemplateAsync(string apimname, string resourceGroup, string singleApiName, List<TemplateResource> apiTemplateResources, List<TemplateResource> productTemplateResources, string policyXMLBaseUrl, string policyXMLSasToken)
```

Comment:
change name would break the code within other files, please keep this not changed